pkg/client/pg_version_check.go

```diff
@@ -0,0 +1,85 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+var minPostgresVersion = version.Must(version.NewVersion("11.0"))
+
+// define some interface to facilitate unit testing
+
+type scanner interface {
+	Scan(...interface{}) error
+}
+
+type connection interface {
+	QueryRow(ctx context.Context, sql string, args ...interface{}) scanner
+	Release()
+}
+
+type pooler interface {
+	Acquire(context.Context) (connection, error)
+}
```

Comment:
I think you can use the QueryExecer interface instead of defining our own interfaces. It should make this code smaller.

Reply:
Can't find that interface. Have a link?

Reply:
https://pkg.go.dev/github.com/jackc/pgtype/pgxtype#Querier